back-dispatcher/routers/EverythingRoute/everythingController.ts

```diff
@@ -0,0 +1,54 @@
+import { Request, Response, NextFunction } from "express";
+import dotenv from "dotenv";
+import {
+  mustIncludeAtLeastOneEverything,
+  unknownError,
+} from "../../constants/strings/strings";
+import { everythingParams } from "../../types/requstParams";
+import { defaultPage, defaultPageSize } from "../../constants/numbers/numbers";
+const NewsAPI = require("newsapi"); // unable to import
+dotenv.config();
+
+export const getEverything = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const {
+      q,
+      sources,
+      sortBy,
+      from,
+      to,
+      language,
+      page,
+      domains,
+      searchIn,
+      excludeDomains,
+      pageSize,
+    } = req.headers as everythingParams;
+
+    if (!q && !sources && !domains) throw mustIncludeAtLeastOneEverything;
+    const newsapiCall = new NewsAPI(process.env.APIKEY);
+    const apiResponse = await newsapiCall.v2.everything({
+      q,
+      sources,
+      domains,
+      language,
+      sortBy,
+      page: page || defaultPage,
+      from,
+      to,
+      searchIn,
+      excludeDomains,
+      pageSize: pageSize || defaultPageSize,
+    });
```

Comment:
1. i want the all params will sent to server via the req body, not the header / params.
    this way the request url string will see more clean.
2. you have interface of the params, use it more elegant way :
cont query : everythingParams  = req.body